### PR TITLE
Python dates

### PR DESCRIPTION
**Pull request by ramanic**
### Description

I have added new entry dates for Python topic.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/python/concepts/dates/dates.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+---
+Title: 'Dates'
+
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:3]: Delete extra newline.

File: content/python/concepts/dates/dates.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+Description: 'Python does not have any data types for date and time but we can work with date and time in python using "datetime" module. It comes built in with python so we do not have to install it but it needs to be imported using "import datetime" statement. It can be used to create & modify datetime objects or to get current datetime. For example .datetime.now() method will return the current date and time.'
+
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:5]: Delete extra newline.

File: content/python/concepts/dates/dates.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+
+Tags:
+  - 'Date'
+  - 'Time'
+
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:13]: Delete extra newline.

File: content/python/concepts/dates/dates.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+---
+Title: 'Dates'
+
+Description: 'Python does not have any data types for date and time but we can work with date and time in python using "datetime" module. It comes built in with python so we do not have to install it but it needs to be imported using "import datetime" statement. It can be used to create & modify datetime objects or to get current datetime. For example .datetime.now() method will return the current date and time.'
+
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:9]: Delete extra newline.

File: content/python/concepts/dates/dates.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+---
+Title: 'Dates'
+
+Description: 'Python does not have any data types for date and time but we can work with date and time in python using "datetime" module. It comes built in with python so we do not have to install it but it needs to be imported using "import datetime" statement. It can be used to create & modify datetime objects or to get current datetime. For example .datetime.now() method will return the current date and time.'
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:4]:
```
Description: 'The built-in Python module called datetime is used to create and modify dates and times.'
```

File: content/python/concepts/dates/dates.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+Tags:
+  - 'Date'
+  - 'Time'
+
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+Python doesn't have a data type for datetime but there is built in module called [datetime](https://docs.python.org/3/library/datetime.html) that can be used to create and modify datetime objects.
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:19]: There's a built-in module called [datetime](https://docs.python.org/3/library/datetime.html) that can be used to create and modify `datetime` objects.

File: content/python/concepts/dates/dates.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+```codebyte/py
+import datetime
+
+current_datetime = datetime.datetime.now()
+
+print(current_datetime)
+```
+
+The result will be the current date and time as below :
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:37]: Delete extra space before the `:`

File: content/python/concepts/dates/dates.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+date_object = datetime.date(2021,10,5)
+# Creates date object with year=2021, month=10 and day=5
+
+time_object = datetime.time(10,5,31)
+# Creates time object with  hour=10,minute=5 and seconds=31
+
+datetime_object=datetime.datetime(2021,10,5,10,5,31)
+# Create datetime object with year=2021, month=10, day=5,hour=10,minute=5 and second=31
+
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:58]: Delete extra line.

File: content/python/concepts/dates/dates.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+import datetime
+
+current_date = datetime.datetime.now()
+#get current datetime
+
+date_after_two_months = current_date + datetime.timedelta(days=30*2)
+
+print(date_after_two_months)
+# Displays date 2 months from current date.
+
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:80]: Delete extra line.

File: content/python/concepts/dates/dates.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+Date and time difference can be calculated using difference operation in `datetime` module.
+
+```codebyte/py
+import datetime
+
+current_date = datetime.datetime.now()
+#get current datetime
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:89]:
```
# Get current datetime
```

File: content/python/concepts/dates/dates.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+## Operations on datetime object
+
+We can perform different operation on a datetime object.
+For example: If you want to add 2 months to a current date , it can be done as below
+
+```codebyte/py
+import datetime
+
+current_date = datetime.datetime.now()
+#get current datetime
```
Comment (sonnynomnom) [inline content/python/concepts/dates/dates.md:74]:
```
# Get current datetime
```

**Review by sonnynomnom — CHANGES_REQUESTED** [review]
Some very minor tweaks to start. Please make the changes so we can move on to review #2!

**Comment by ramanic** [comment]
Please check, I have made the changes.

**Comment by sonnynomnom** [comment]
@yangc95 @Dusch4593 

I made a quick pass. One of you should give it a more thorough review tomorrow!

File: content/python/concepts/dates/dates.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  - 'Date'
+  - 'Time'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+Python doesn't have a data type for datetime but there's a built-in module called [datetime](https://docs.python.org/3/library/datetime.html) that can be used to create and modify datetime objects.
+
+The module comes built in with python so installation is not required but we need to import it at top of Python file as below:
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:17]:
```suggestion
The module comes built-in with Python so installation is not required, but we do need to import it at top of a Python file as below:
```

File: content/python/concepts/dates/dates.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+Tags:
+  - 'Date'
+  - 'Time'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+Python doesn't have a data type for datetime but there's a built-in module called [datetime](https://docs.python.org/3/library/datetime.html) that can be used to create and modify datetime objects.
+
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:16]:
```suggestion
```

File: content/python/concepts/dates/dates.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+current_datetime = datetime.datetime.now()
+
+print(current_datetime)
+```
+
+The result will be the current date and time as below:
+
+`2021-10-05 10:20:57.944968`
+
+It contains year,month,day,hour,minute,second and microsecond.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:37]:
```suggestion
It contains the year, month, day, hour, minute, second and microsecond.
```

File: content/python/concepts/dates/dates.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+## Creating new datetime object
+
+`datetime` module have three main available types i.e `date` , `time` and `datetime`.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:41]:
```suggestion
The `datetime` module has three main available types: `date` , `time` and `datetime`.
```

File: content/python/concepts/dates/dates.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+import datetime
+
+date_object = datetime.date(2021,10,5)
+# Creates date object with year=2021, month=10 and day=5
+
+time_object = datetime.time(10,5,31)
+# Creates time object with  hour=10,minute=5 and seconds=31
+
+datetime_object=datetime.datetime(2021,10,5,10,5,31)
+# Create datetime object with year=2021, month=10, day=5,hour=10,minute=5 and second=31
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:53]:
```suggestion
# Creates datetime object with year=2021, month=10, day=5, hour=10, minute=5 and second=31
```

File: content/python/concepts/dates/dates.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+Further more we also have a timedelta type which is used to represent time difference.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:56]:
```suggestion
Furthermore, we also have a timedelta type which is used to represent time difference.
```

File: content/python/concepts/dates/dates.md [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+`timedelta(days, seconds, microseconds, milliseconds, minutes, hours, weeks)`
+All arguments are optional and have default value of 0.
+
+## Operations on datetime object
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:60]:
```suggestion
## Operations on a datetime Object
```

File: content/python/concepts/dates/dates.md [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+
+Python doesn't have a data type for datetime but there's a built-in module called [datetime](https://docs.python.org/3/library/datetime.html) that can be used to create and modify datetime objects.
+
+The module comes built in with python so installation is not required but we need to import it at top of Python file as below:
+
+```py
+import datetime
+```
+
+For example `.datetime.now()` method will return the current date and time.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:23]:
```suggestion
A basic example is employing the `.datetime.now()` method to return the current date and time:
```

File: content/python/concepts/dates/dates.md [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+time_object = datetime.time(10,5,31)
+# Creates time object with  hour=10,minute=5 and seconds=31
+
+datetime_object=datetime.datetime(2021,10,5,10,5,31)
+# Create datetime object with year=2021, month=10, day=5,hour=10,minute=5 and second=31
+```
+
+Further more we also have a timedelta type which is used to represent time difference.
+`timedelta(days, seconds, microseconds, milliseconds, minutes, hours, weeks)`
+All arguments are optional and have default value of 0.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:58]:
```suggestion
Note that all of the arguments are optional and have default value of 0.
```

File: content/python/concepts/dates/dates.md [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+
+## Operations on datetime object
+
+We can perform different operation on a datetime object.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:62]:
```suggestion
We can perform different operations on a datetime object.
```

File: content/python/concepts/dates/dates.md [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+```codebyte/py
+import datetime
+
+current_date = datetime.datetime.now()
+#Get current datetime
+
+date_after_two_months = current_date + datetime.timedelta(days=30*2)
+
+print(date_after_two_months)
+# Displays date 2 months from current date.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:74]:
```suggestion
# Displays date 2 months from current date
```

File: content/python/concepts/dates/dates.md [hunk 23]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+Date and time difference can be calculated using difference operation in `datetime` module.
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:77]:
```suggestion
Date and time difference can be calculated using the difference or subtraction operator on `datetime` objects.
```

File: content/python/concepts/dates/dates.md [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+
+```codebyte/py
+import datetime
+
+current_date = datetime.datetime.now()
+#Get current datetime
```
Comment (caupolicandiaz) [inline content/python/concepts/dates/dates.md:83]:
```suggestion
#Get current datetime
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
I've made some suggested changes for readability and consistency, but overall many of the edits are spacing and punctuation.

**Comment by ramanic** [comment]
@caupolicandiaz Thanks, I have committed the chages.

**Review by sonnynomnom — COMMENTED** [review]
Some final changes!

**Comment by sonnynomnom** [comment]
Merged! Congrats!

I made some more changes here btw: https://github.com/Codecademy/docs/blob/main/content/python/concepts/dates/dates.md

cc @caupolicandiaz @ramanic

**Comment by ramanic** [comment]
Cool thanks :), will love to contribute more.

**Pull request merged**